framework/include/variables/MooseVariableBase.h

```diff
@@ -178,7 +181,7 @@ class MooseVariableBase : public MooseObject,
   /**
    * @return whether this is an array variable
    */
-  bool isArray() const { return _is_array; }
+  virtual bool isArray() const { return _is_array; }
```

Comment:
Why? And can't this just use `_array_var_component_names.size()` now?

Reply:
No because an array variable is allowed to have only one component.

Reply:
In this case, I can actually just check if `_array_var_component_names` is empty or not. If the array variable has a single component, `_array_var_component_names` will still have a single name in it. It is only empty if the `array` parameter was false

Reply:
Oh, sorry, I see.